File: src/SOP8/Egovframework/Base/base-account-ibll/src/main/java/egovframework/base/account/ibll/models/ApplicationUser.java
package egovframework.base.account.ibll.models;

import egovframework.base.model.account.User;
import egovframework.base.model.account.Grade;
import com.fasterxml.jackson.annotation.JsonIgnore;

public class ApplicationUser {

    private int userSn = -1;
    private int gradSn = -1;
    private String gradName = "";
    private String userId = "";
    private String userName = "";
    private String sessionKey = "";
    @JsonIgnore
    private Object options = new Object();
    private Integer siteSn = null;
    private Integer regularMemberNo = null;

    public int getUserSn() {
        return userSn;
    }

    public void setUserSn(int userSn) {
        this.userSn = userSn;
    }

    public int getGradSn() {
        return gradSn;
    }

    public void setGradSn(int gradSn) {
        this.gradSn = gradSn;
    }

    public String getGradName() {
        return gradName;
    }

    public void setGradName(String gradName) {
        this.gradName = gradName;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getSessionKey() {
        return sessionKey;
    }

    public void setSessionKey(String sessionKey) {
        this.sessionKey = sessionKey;
    }

    public Object getOptions() {
        return options;
    }

    public void setOptions(Object options) {
        this.options = options;
    }

    public Integer getSiteSn() {
        return siteSn;
    }

    public void setSiteSn(Integer siteSn) {
        this.siteSn = siteSn;
    }

    public Integer getRegularMemberNo() {
        return regularMemberNo;
    }

    public void setRegularMemberNo(Integer regularMemberNo) {
        this.regularMemberNo = regularMemberNo;
    }

    public static ApplicationUser makeUser(User user, Grade grade, String strSessionKey) {
        ApplicationUser appUser = new ApplicationUser();
        appUser.setUserSn(user.getUser_sn());
        appUser.setGradSn(user.getGrad_sn());
        appUser.setGradName(grade.getGrad_name());
        appUser.setUserId(user.getUser_id());
        appUser.setUserName(user.getUser_name());
        appUser.setSessionKey(strSessionKey);
        appUser.setSiteSn(user.getSite_sn());
        appUser.setRegularMemberNo(user.getRgl_memb_sn());
        return appUser;
    }
}

File: src/SOP8/Base/Account/Base.Account.IBLL/Models/ApplicationUser.cs
using Base.Model.Account;

namespace Base.Account.IBLL.Models
{
    public class ApplicationUser
    {
        private int m_nNo = -1;
        private int m_nLevelNo = -1;
        private string m_strLevel = "";
        private string m_strUserID = "";
        private string m_strNickName = "";
        private string m_strSessionKey = "";
        private object m_options = new object();
        private int? m_siteNo = null;
        private int? m_regularMemberNo = null;

        public int user_sn
        {
            get { return m_nNo; }
            set { m_nNo = value; }
        }

        public int grad_sn
        {
            get { return m_nLevelNo; }
            set { m_nLevelNo = value; }
        }

        public string grad_name
        {
            get { return m_strLevel; }
            set { m_strLevel = value; }
        }

        public string user_id
        {
            get { return m_strUserID; }
            set { m_strUserID = value; }
        }

        public string user_name
        {
            get { return m_strNickName; }
            set { m_strNickName = value; }
        }

        public string session_key
        {
            get { return m_strSessionKey; }
            set { m_strSessionKey = value; }
        }

        public object Options
        {
            get { return m_options; }
            set { m_options = value; }
        }

        public int? site_sn
        {
            get { return m_siteNo; }
            set { m_siteNo = value; }
        }

        public int? RegularMemberNo
        {
            get { return m_regularMemberNo; }
            set { m_regularMemberNo = value; }
        }

        public static ApplicationUser MakeUser(User user, Grade grade, string strSessionKey)
        {
            ApplicationUser appUser = new ApplicationUser();
            appUser.user_sn = user.user_sn;
            appUser.grad_sn = user.grad_sn;
            appUser.grad_name = grade.grad_name;
            appUser.user_id = user.user_id;
            appUser.user_name = user.user_name;
            appUser.session_key = strSessionKey;
            appUser.site_sn = user.site_sn;
            appUser.RegularMemberNo = user.rgl_memb_sn;

            return appUser;
        }
    }
}
